unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs

```diff
@@ -0,0 +1,309 @@
+﻿/**
+ * Copyright 2018-2020 MobiledgeX, Inc. All rights and licenses reserved.
+ * MobiledgeX, Inc. 156 2nd Street #408, San Francisco, CA 94105
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using UnityEngine;
+
+public class FaceDetectionRects : Stopwatch
+{
+  string TAG = "FaceDetectionRects";
+  List<Rect> rects = new List<Rect>();
+  private TimeSpan TTL = new System.TimeSpan(0, 0, 0, 4);
+
+  public delegate void DetectionCallback();
+  public DetectionCallback consumers;
+
+  Vector3 position;
+  Quaternion rotation;
+
+  object syncObject;
+
+  public FaceDetectionRects()
+  {
+    syncObject = new object();
+    consumers = UnWatched;
+  }
+
+  private void ClearRects()
+  {
+    rects.Clear();
+  }
+
+  private void UnWatched()
+  {
+    int? num;
+    num = consumers?.GetInvocationList().Length;
+
+    if (num == 1)
+    {
+      Logger.D(TAG, "Default no-op consumer notified.");
+    }
+    else
+    {
+      Logger.D(TAG, "Num consumers: " + num);
+    }
+  }
+
+  public void Update(List<Rect> newRects, Vector3 position, Quaternion rotation)
+  {
+    lock (syncObject)
+    {
+      Reset();
+      if (newRects == null)
+      {
+        this.rects.Clear();
+      }
+      else
+      {
+        rects = newRects;
+      }
+      this.position = new Vector3(position.x, position.y, position.z);
+      this.rotation = new Quaternion(rotation.x, rotation.y, rotation.z, rotation.w);
+      consumers();
+    }
+
+  }
+
+  public bool Expired()
+  {
+    if (Elapsed > TTL)
+    {
+      return true;
+    }
+    return false;
+  }
+
+  public bool DoExpire()
+  {
+    if (Expired())
+    {
+      ClearRects();
+      Reset();
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Auto expires, then return rects;
+   */
+  public List<Rect> GetRects()
+  {
+    lock (syncObject)
+    {
+      DoExpire();
+    }
+    return rects;
+  }
+
+}
+
+public class FaceDetectionUI : MonoBehaviour
+{
+  const string TAG = "FaceDetectionUI";
+  GameObject faceDetectionGo;
+
+  public delegate void DetectionCallback(int numDetected);
+  DetectionCallback aDetectionCallback;
```

Comment:
Are you using aDetectionCallback somewhere?

Reply:
Nope, not in this code, removed.